Lib/cu2qu/ufo.py

```diff
@@ -0,0 +1,198 @@
+
+    def _add_segment(self, tag, *args):
+        if tag in ['move', 'line', 'qcurve', 'curve']:
+            self._last_pt = args[-1]
+        self.segments.append((tag, args))
+
+    def moveTo(self, pt):
+        self._add_segment('move', pt)
+
+    def lineTo(self, pt):
+        self._add_segment('line', pt)
+
+    def qCurveTo(self, *points):
+        self._add_segment('qcurve', self._last_pt, *points)
+
+    def curveTo(self, *points):
+        self._add_segment('curve', self._last_pt, *points)
+
+    def closePath(self):
+        self._add_segment('close')
+
+    def endPath(self):
+        self._add_segment('end')
+
+    def addComponent(self, glyphName, transformation):
+        self._add_segment('component', glyphName, transformation)
+
+
+def _get_segments(glyph):
+    """Get a glyph's segments as extracted by GetSegmentsPen."""
+
+    pen = GetSegmentsPen()
+    glyph.draw(pen)
+    return pen.segments
+
+
+def _set_segments(glyph, segments):
+    """Draw segments as extracted by GetSegmentsPen back to a glyph."""
+
+    glyph.clearContours()
+    pen = glyph.getPen()
+    for tag, args in segments:
+        if tag == 'move':
+            pen.moveTo(*args)
+        elif tag == 'line':
+            pen.lineTo(*args)
+        elif tag == 'qcurve':
+            pen.qCurveTo(*args[1:])
+        elif tag == 'close':
+            pen.closePath()
+        elif tag == 'end':
+            pen.endPath()
+        elif tag == 'component':
+            pen.addComponent(*args)
+        else:
+            raise AssertionError('Unhandled segment type "%s"' % tag)
+
+
+def _segments_to_quadratic(segments, max_err, stats):
+    """Return quadratic approximations of cubic segments."""
+
+    assert all(s[0] == 'curve' for s in segments), 'Non-cubic given to convert'
+
+    new_points, _ = curves_to_quadratic([s[1] for s in segments], max_err)
+    n = len(new_points[0])
+    assert all(len(s) == n for s in new_points[1:]), 'Converted incompatibly'
+
+    n = str(n)
+    if stats is not None:
+        stats[n] = stats.get(n, 0) + 1
+
+    return [('qcurve', p) for p in new_points]
+
+
+def _fonts_to_quadratic(fonts, max_err, stats):
+    """Do the actual conversion of fonts, after arguments have been set up."""
+
+    for glyphs in zip(*fonts):
+        name = glyphs[0].name
+        assert all(g.name == name for g in glyphs), 'Incompatible fonts'
+
+        segments_by_location = zip(*[_get_segments(g) for g in glyphs])
+        if not any(segments_by_location):
+            continue
+
+        new_segments_by_location = []
+        for segments in segments_by_location:
+            tag = segments[0][0]
+            assert all(s[0] == tag for s in segments[1:]), (
+                'Incompatible glyphs "%s"' % name)
+            if tag == 'curve':
+                segments = _segments_to_quadratic(segments, max_err, stats)
+            new_segments_by_location.append(segments)
+
+        new_segments_by_glyph = zip(*new_segments_by_location)
```

Comment:
in Python 3, `zip` returns an iterator instead of a list like in Python 2.
When you are looping through the zip results more than once, you need to convert them into a list, otherwise the iterator will return empty the second time.

Besides, because in your `zip` implementation you are checking `len(a) for a in args`, if one of the arguments does not support `__len__` because it is in turn an iterator object from a previous `zip` call, then a `TypeError: object of type 'zip' has no len()` is raised. So you need to convert iterator arguments to lists (which have `len()`) before calling zip on them. 
 
```diff
diff --git a/Lib/cu2qu/ufo.py b/Lib/cu2qu/ufo.py
index c4d11d2..12c0f4b 100644
--- a/Lib/cu2qu/ufo.py
+++ b/Lib/cu2qu/ufo.py
@@ -135,7 +135,7 @@ def _fonts_to_quadratic(fonts, max_err, stats):
         name = glyphs[0].name
         assert all(g.name == name for g in glyphs), 'Incompatible fonts'
 
-        segments_by_location = zip(*[_get_segments(g) for g in glyphs])
+        segments_by_location = list(zip(*[_get_segments(g) for g in glyphs]))
         if not any(segments_by_location):
             continue
 
@@ -148,7 +148,7 @@ def _fonts_to_quadratic(fonts, max_err, stats):
                 segments = _segments_to_quadratic(segments, max_err, stats)
             new_segments_by_location.append(segments)
 
-        new_segments_by_glyph = zip(*new_segments_by_location)
+        new_segments_by_glyph = list(zip(*new_segments_by_location))
         for glyph, new_segments in zip(glyphs, new_segments_by_glyph):
             _set_segments(glyph, new_segments)
```